src/app/core/header/timepiece/timepiece.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Observable } from 'rxjs/Rx';
+import { Subscription } from 'rxjs';
+
+@Component({
+    selector: 'sh-timepiece',
+    styleUrls: ['./timepiece.component.scss'],
+    templateUrl: './timepiece.component.html'
+})
+
+export class TimepieceComponent implements OnInit, OnDestroy {
+
+    private currentTime: Date;
+    private timer: Observable<Object>;
+    private timeTracker: Subscription;
+
+    ngOnInit() {
+        this.createTimer(0, 1000);
```

Comment:
Your solution looks quite good, and as for me I agree with this, and in that time it looks a bit complicated. 

We can make it simplier with usage just
```
this.timer = Observable
            .interval(period)
            .map(() => new Date());
```
in ```ngOnInit``` instead of other

and in html template use 
```
{{timer | async | date:'MM.dd.y HH:mm:ss'}}
```
What you think about this?